game/resources/sfx.lua

```diff
@@ -0,0 +1,49 @@
+local Sfx = {}
+
+local Source = Class{}
+
+function Source:init(path, polyphony)
+    assert(polyphony > 0, "Not a valid polyphony value")
+    self.buffer_size = polyphony
+
+    self.buffer = {}
+    for i = 1, self.buffer_size do
+      self.buffer[i] = love.audio.newSource(path, "static")
+    end
+
+    self.next_index = 1
+end
+
+function Source:play()
+    self.buffer[self.next_index]:play()
+    self.next_index = (self.next_index % self.buffer_size) + 1
+    return self:getLast()
+end
+
+function Source:getLast()
+    return self.buffer[((self.next_index + self.buffer_size - 2) % self.buffer_size) + 1]
+end
+
+function Source:setVolume(vol)
+  for i = 1, self.buffer_size do
+    self.buffer[i]:setVolume(vol)
+  end
+end
+
+function Source:seek(vol)
+  for i = 1, self.buffer_size do
+    self.buffer[i]:setVolume(vol)
+  end
+end
+
+function Source:getVolume()
+    return self.buffer[1]:getVolume()
+end
+
+---------------------------------
+
+function Sfx.new(path, polyphony)
+  return Source(path, polyphony)
+end
```

Comment:
Why the wrapper? Is this a new convention?

Reply:
I just left room in case we wanted other functions for this module, instead of just returning the Source class (I was initially going to return the Class directly, but other resource files such as sprite.lua have other functions, so that's my reasoning in "wrapping" haha)

Reply:
I'm sorry but I didn't understand that. Could you try phrasing it differently?

Reply:
In case we want to add other auxiliary functions to this module, I'm not returning the Source class directly. That's why there is a function to create a Source and return it.

Reply:
oh ok